src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -306,6 +306,15 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                                         foreach (IParameterSymbol paramSymbol in methodSymbol.Parameters)
                                         {
                                             string paramName = paramSymbol.Name;
+                                            bool needsAtSign = false;
+                                            if (paramSymbol.DeclaringSyntaxReferences.Length > 0)
+                                            {
+                                                ParameterSyntax paramSyntax = paramSymbol.DeclaringSyntaxReferences[0].GetSyntax(_cancellationToken) as ParameterSyntax;
+                                                if (paramSyntax != null && !string.IsNullOrEmpty(paramSyntax.Identifier.Text))
+                                                {
+                                                    needsAtSign = paramSyntax.Identifier.Text[0] == '@';
+                                                }
+                                            }
```

Comment:
Wouldn't this simpler approach work?

```suggestion
                                            bool needsAtSign = SyntaxFacts.GetKeywordKind(paramName) != SyntaxKind.None;;
```

Reply:
@chsienki @333fred

Reply:
Tested out the commit suggestion and seems to work fine. Wondering which one is a more robust approach here.

Reply:
This isn't going to be fully robust, particularly in the light of things like the lowercase keyword warning we've added. I don't think this would say that `record` or `required` would need to be escaped, for example.